executor/executor.go

```diff
@@ -83,6 +87,101 @@ type Executor interface {
 	Close() error
 }
 
+// ShowDDL represents a show DDL executor.
+type ShowDDL struct {
+	fields []*ast.ResultField
+	ctx    context.Context
+	cnt    int
+}
+
+// Fields implements Executor Fields interface.
+func (e *ShowDDL) Fields() []*ast.ResultField {
+	return e.fields
+}
+
+// Next implements Execution Next interface.
+func (e *ShowDDL) Next() (*Row, error) {
+	if e.cnt != 0 {
+		return nil, nil
+	}
+
+	txn, err := e.ctx.GetTxn(false)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	info, err := inspectkv.GetDDLInfo(txn)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	records := []interface{}{
+		info.SchemaVer,
+		nil,
+		nil,
+	}
+	if info.Owner != nil {
+		records[1] = info.Owner.String()
+	}
+	if info.Job != nil {
+		records[2] = info.Job.String()
+	}
+	row := &Row{}
+	row.Data = append(row.Data, records...)
+	for i, f := range e.fields {
+		f.Expr.SetValue(records[i])
+	}
+	e.cnt++
+
+	return row, nil
+}
+
+// Close implements plan.Plan Close interface.
+func (e *ShowDDL) Close() error {
+	return nil
+}
+
+// CheckTable represents a check table executor.
+type CheckTable struct {
+	tables []*ast.TableName
+	ctx    context.Context
+}
+
+// Fields implements Executor Fields interface.
+func (e *CheckTable) Fields() []*ast.ResultField {
+	return nil
+}
+
+// Next implements Execution Next interface.
+func (e *CheckTable) Next() (*Row, error) {
```

Comment:
There should be a `e.count` or `e.done`  field to prevent executing more than once.